packages/statemanager/test/testdata/providerData/mockProvider.ts

```diff
@@ -1,122 +1,94 @@
-import { JsonRpcProvider, Network } from 'ethers'
+import type { JsonBlock } from '@ethereumjs/block'
 
-import type { FetchRequest, JsonRpcPayload, JsonRpcResult } from 'ethers'
+export type SupportedMethods =
+  | 'eth_getProof'
+  | 'eth_getStorageAt'
+  | 'eth_getCode'
+  | 'eth_getBlockByNumber'
+  | 'eth_getTransactionByHash'
 
-export class MockProvider extends JsonRpcProvider {
-  constructor() {
-    super('localhost:8545', Network.from('mainnet'), { staticNetwork: Network.from('mainnet') })
-  }
-
-  _getConnection(): FetchRequest {
-    const fakeConnection = {
-      url: 'localhost:8545',
-    }
-    return fakeConnection as FetchRequest
-  }
-
-  _send = async (payload: JsonRpcPayload | JsonRpcPayload[]): Promise<JsonRpcResult[]> => {
-    let method
-    let params
-    let id
-
-    if (Array.isArray(payload)) {
-      const results = []
-      for (const el of payload) {
-        ;({ method, params, id } = el)
-        results.push(await this.getValues(method, id, params))
+export type JsonReturnType = {
+  eth_getProof: { id: number; result: any }
+  eth_getStorageAt: { id: number; result: any }
+  eth_getCode: { id: number; result: string }
+  eth_getBlockByNumber: { id: number; result: JsonBlock }
+  eth_getTransactionByHash: { id: number; result: any }
+}
+export const getValues = async <Method extends SupportedMethods>(
```

Comment:
Not sure here, what is this `Method` generic type? Shouldn't this be `SupportedMethods` as type itself?

Reply:
Oh right this makes sense, you need it for `method` param